src/type-coercions.md

```diff
@@ -211,6 +215,13 @@ r[coerce.unsize.slice]
 r[coerce.unsize.trait-object]
 * `T` to `dyn U`, when `T` implements `U + Sized`, and `U` is [dyn compatible].
 
+r[coerce.unsize.trait-upcast]
```

Comment:
I think this whole section might need a retractor :p

Specifically I think it should more clearly state the difference between `Unsize` and `CoerceUnsize`, mention that the former is not actually a coercion, and the latter is only for pointers. 

Then it would also make sense to add a note that unsizing coercion changes the metadata of a pointer, while all other coercions/casts don't.

But, if you don't mind I'll leave this as a follow up ^^'